src/main/java/de/baleipzig/iris/common/utils/LayerUtils.java

```diff
@@ -23,58 +24,78 @@ public static int getNumberOfNodes(ILayer layer){
         int sizeY = layer.getDim().getY();
         List<INode> list = new ArrayList<>(sizeX * sizeY);
 
-        for(int y = 0; y < sizeY; y++)
-            for(int x = 0; x < sizeX; x++){
-                list.add(layer.getNode(x,y));
+        for(int y = 0; y < sizeY; y++) {
+            for (int x = 0; x < sizeX; x++) {
+                list.add(layer.getNode(x, y));
             }
+        }
 
         return list;
     }
 
-    public static ILayer createLayer(Dimension dim, boolean useRandomBias){
+    public static void copyStatesFromTo(ILayer sourceLayer, ILayer copyLayer) {
+
+        final Dimension sourceDim = sourceLayer.getDim();
+        final Dimension copyDim = copyLayer.getDim();
+
+        if( copyDim.getY() != sourceDim.getY() || copyDim.getX() != sourceDim.getX()) {
```

Comment:
ne equals methode in Dimension fänd ich gut (gibts vielleicht geschenkt mit lombok, @data oder @equalsAndHashcode)